### PR TITLE
Handle protection status errors for unprotected branches

### LINKED ISSUE #625: Branch protection status checks returns error for unprotected branches
Response is 404 so throws an error, instead of returning nil slice of required contexts. Sending in a quick PR.

### PR DESCRIPTION
**Pull request by hemachandarv**
Resolves #625

### CODE REVIEW

**Comment by hemachandarv** [comment]
@gmlewis I've addressed #625 by returning an explicit error in the `GET` APIs. Do you think this is a good idea?

Also, do you think we should cover the same for the `POST` APIs too?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @hemachandarv !
Just a few minor tweaks, please.

File: github/repos_test.go [hunk 1]
```diff
@@ -1150,6 +1150,32 @@ func TestRepositoriesService_GetBranchProtection_noDismissalRestrictions(t *test
...
+func TestRepositoriesService_GetBranchProtection_branchNotProtected(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/branches/b/protection", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, `{
+			"message": "Branch not protected",
```
[the remark this answers is not in the document]
Comment (hemachandarv) [reply github/repos_test.go:1162]: @gmlewis I'm not sure if we'll be able to use a variable inside a raw string. Am I missing something..?

File: github/repos.go [hunk 2]
```diff
@@ -1539,3 +1553,12 @@ func (s *RepositoriesService) Dispatch(ctx context.Context, owner, repo string,
...
 	return r, resp, nil
 }
+
+// isBranchNotProtected determines whether a branch is not protected
+// based on the error message returned by GitHub API.
+func isBranchNotProtected(err error) bool {
+	if errorResponse, ok := err.(*ErrorResponse); ok {
+		return errorResponse.Message == githubBranchNotProtected
+	}
+	return false
```
[the remark this answers is not in the document]
Comment (hemachandarv) [reply github/repos.go:1563]: @gmlewis Thank you for suggesting this change. I certainly missed this simplification.

File: github/repos_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/repos_test.go:1162]: Whups, you are correct. Please use `%v` and the variable name in that case.

[the remark this answers is not in the document]
Comment (hemachandarv) [reply github/repos_test.go]: Thanks @gmlewis. I've updated the PR with the suggested changes.

**Review by gmlewis — APPROVED** [review]
Thank you, @hemachandarv !
LGTM.

Awaiting second LGTM before merging.

**Review by cpanato — APPROVED** [review]
LGTM
thanks!

**Comment by gmlewis** [comment]
Thank you, @cpanato !
Merging.

**Pull request merged**